azure-pipelines.yml

```diff
@@ -2,72 +2,97 @@ trigger:
   branches:
     include:
     - refs/tags/v*
+    
+jobs:
+- job: Build
+  pool:
+    name: Azure-IoT-EdgeExperience-1ES-Hosted-Linux
+    demands:
+      - ImageOverride -equals MMSUbuntu18.04TLS
 
-pool:
-  name: Azure-IoT-EdgeExperience-1ES-Hosted-Linux
-  demands:
-    - ImageOverride -equals MMSUbuntu18.04TLS
+  steps:
+    - task: NodeTool@0
+      inputs:
+        versionSpec: '14.x'
+        checkLatest: true
 
-steps:
-  - task: NodeTool@0
-    inputs:
-      versionSpec: '14.x'
-      checkLatest: true
+    - script: |
+          npm install yo
+          npm install 
+      displayName: 'npm install'
+      
+    - script: |
+          npm run lint 
+      displayName: 'npm lint'
 
-  - script: |
-        npm install yo
-        npm install 
-    displayName: 'npm install'
-    
-  - script: |
-        npm run lint 
-    displayName: 'npm lint'
+    - task: Npm@1
+      displayName: "npm Packaging"
+      inputs:
+          command: 'custom'
+          customCommand: 'pack'
 
-  - task: Npm@1
-    displayName: "npm Packaging"
-    inputs:
-        command: 'custom'
-        customCommand: 'pack'
+    - task: AzureArtifacts.manifest-generator-task.manifest-generator-task.ManifestGeneratorTask@0
+      displayName: 'SBOM Generation Task'
+      inputs:
+        BuildDropPath: '$(Build.ArtifactStagingDirectory)'
 
-  - task: AzureArtifacts.manifest-generator-task.manifest-generator-task.ManifestGeneratorTask@0
-    displayName: 'SBOM Generation Task'
-    inputs:
-      BuildDropPath: '$(Build.ArtifactStagingDirectory)'
+    - task: Bash@3
+      inputs:
+        targetType: 'inline'
+        script: |
+          if [[ $(Build.SourceBranch) =~ ^refs/tags/v?[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+          echo "##vso[task.setvariable variable=NPM_PUBLISH_FLAG;]true"
+          fi
 
-  - task: CopyFiles@2
-    inputs:
-      SourceFolder: '$(Build.SourcesDirectory)'
-      Contents: '*.tgz'
-      TargetFolder: '$(Build.ArtifactStagingDirectory)/Package'
-  - task: PublishBuildArtifacts@1
-    inputs:
-      PathtoPublish: '$(Build.ArtifactStagingDirectory)/Package'
-      ArtifactName: 'drop'
-      publishLocation: 'Container'
+    - task: Bash@3
+      condition: and(succeeded(), eq(variables['NPM_PUBLISH_FLAG'], 'true'))
+      inputs:
+        targetType: 'inline'
+        script: |
+          VERSION=$(node -p -e "require('./package.json').version")
+          echo "##vso[task.setvariable variable=PACKAGEVERSION;]$VERSION"
```

Comment:
Let's move this block with the rest of npm publishing tasks, maybe l.80? This way the npm logic will be collocated.

Reply:
We are using the variable $(PACKAGEVERSION) in GitHub release we can move it to line l.64.

Reply:
Good catch, i missed that. Thanks